src/helpers/validator.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Creates a user-type guard for objects provided a simple key-to-type schema.
+ * Optional (?) types should not be provided in the simple schema.
```

Comment:
Are we planning on adding another method for type validation for other optional keys?

Reply:
i would like to, except that inherently an optional key of type T is `T | undefined`
a different type of validator could be instead of `key: 'type'` it would be `key: predicate`
this way we can support chaining (e.g. `key: optionalKey(isString)` would match `key?: string`)

but I'd if we want better validator, it might be worth just finding a battle-tested validator library instead of making one ourselves

Reply:
haha, yeah. You're right, I'd imagine that this is a common problem